web/cobrands/highways/assets.js

```diff
@@ -0,0 +1,121 @@
+(function(){
+
+if (!fixmystreet.maps) {
+    return;
+}
+
+var defaults = {
+    http_options: {
+        url: "https://tilma.mysociety.org/mapserver/highways",
+        params: {
+            SERVICE: "WFS",
+            VERSION: "1.1.0",
+            REQUEST: "GetFeature",
+            SRSNAME: "urn:ogc:def:crs:EPSG::3857"
+        }
+    },
+    format_class: OpenLayers.Format.GML.v3.MultiCurveFix,
+    asset_type: 'area',
+    // this covers zoomed right out on Cumbrian sections of
+    // the M6
+    max_resolution: 20,
+    min_resolution: 0.5971642833948135,
+    srsName: "EPSG:900913",
+    strategy_class: OpenLayers.Strategy.FixMyStreet
+};
+
+var highways_stylemap = new OpenLayers.StyleMap({
+    'default': new OpenLayers.Style({
+        fill: false,
+        stroke: false,
+    })
+});
+
+fixmystreet.assets.add($.extend(true, {}, defaults, {
+    http_options: {
+        params: {
+            TYPENAME: "Highways"
+        }
+    },
+    stylemap: highways_stylemap,
+    always_visible: true,
+
+    non_interactive: true,
+    road: true,
+    all_categories: true,
+    // motorways are wide and the lines to define them are narrow so we
+    // need a bit more margin for error in finding the nearest to stop
+    // clicking in the middle of them being undetected
+    nearest_radius: 15,
+    actions: {
+        found: function(layer, feature) {
+            // if we've changed location then we want to reset things otherwise
+            // this is probably just being called again by a category change
+            var lat = $('#fixmystreet\\.latitude').val(),
+                lon = $('#fixmystreet\\.longitude').val();
+            if ( fixmystreet.body_overrides.location &&
+                 lat == fixmystreet.body_overrides.location.latitude &&
+                 lon == fixmystreet.body_overrides.location.longitude ) {
+                return;
+            }
+            $('#highways').remove();
+            if ( !fixmystreet.assets.selectedFeature() ) {
+                fixmystreet.body_overrides.only_send('Highways England');
+                add_highways_warning(feature.attributes.ROA_NUMBER);
+            }
+        },
+        not_found: function(layer) {
+            fixmystreet.body_overrides.location = null;
+            fixmystreet.body_overrides.remove_only_send();
```

Comment:
When this runs after a Bucks road has been selected with Flytipping already picked, the only-send-to-Bucks is lost because it is always removed by this function. I think this needs to not do this line if only_send is already set to something other than Highways England, does that work?